UKHomeOffice/nmsw-ui
# NMSW-140 Extend conditional input with error handling

# Ticket

NMSW-140

----

## AC

Given I can see the connected text input
When I click submit
And I have not entered anything in the connected field
And the connected field is required
Then I should see an error message for this connected field

GIVEN a connected field has an error
WHEN a click on the error link in the error summary for that field
THEN the page should scroll to the question
AND the connected field should be in focus

Given I can see an error for the connected field
When I enter something in the connected field
Then the error should clear

GIVEN there are no other errors
When I click submit
Then the form should submit as normal

----

## To test

- run app locally
- go to second page
- enter something in first name
- select a colour
- submit form
- > animal question should error
- select `cat`
- submit form
- > breed of cat should error
- type something in `breed of cat` field
- select `dog`
- submit form
- > breed of dog should error 
- type something in `breed of dog` field
- submit form
- form should submit


----

## Notes
